### PR TITLE
Create pretty table

### PR DESCRIPTION
**Pull request by mzayeddfe**
# Brief overview of changes

Created a function called pretty_table

## Why are these changes being made?

To enable the use of pretty_num with tables. 

## Detailed description of changes

- The function is applies to all columns regardless of whether they are numeric or not. 
    - This is so it doesn't exclude columns that have numbers and character/NA values. 
    - This makes it more versatile. 

- It applies the customisation of pretty_num to the inputted table. 

- Users have the option to include and or exclude columns. 
   - If both are chose, the columns included take precedence. 

## Issue ticket number/s and link

[87](https://github.com/dfe-analytical-services/dfeR/issues/87)

### CODE REVIEW

File: tests/testthat/test-pretty_table.R
Comment (cjrace) [inline tests/testthat/test-pretty_table.R:1]: Love this, nice use of a test data frame and some good test cases. Only thing to add would be if you add the validation into it, you could test that the error is thrown if it's not a data.frame object.

Examples of the kind of thing you could use from my testing of this branch locally (assuming the error message will be a nicer custom one we add!)
image

**Review by cjrace — CHANGES_REQUESTED** [review]
Nice, I've made a few small suggestions around validation / naming but having poked it a bit locally all seems to work well!

I think this should be a minor version update on the package, as it's a new feature being added - have a look at the `usethis::use_version()` function to bump the version number and add notes into the news.md file (also, thinking of package maintenance moving forwards it might be worth having a read about [semver versioning](https://semver.org/) more widely if you've not come across it before).

File: R/pretty.R [hunk 1]
```diff
@@ -313,3 +314,92 @@ pretty_num <- function(
...
+#'
+#' # Apply formatting to only selected columns
+#' pretty_table(df, include_columns = c("a"), dp = 2)
+#'
+#' # Apply formatting to all columns except specified ones
+#' pretty_table(df, exclude_columns = c("b"), dp = 2)
+#'
+#' # Apply formatting to all columns except specified ones and
+#' # provide alternative value for NAs
+#' pretty_table(df, alt_na = "[z]", exclude_columns = c("b"), dp = 2)
```
[the remark this answers is not in the document]
Comment (mzayeddfe) [reply R/pretty.R:367]: it was the way we were coding suppressed figures when i was producing tidy data for the hmrc version of ees. i think it just stuck in my head.

File: DESCRIPTION [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 Authors@R: c(
     person("Cam", "Race", , "cameron.race@education.gov.uk", role = c("aut", "cre")),
     person("Laura", "Selby", , "laura.selby@education.gov.uk", role = "aut"),
     person("Adam", "Robinson", role = "aut"),
     person("Jen", "Machin", , "jen.machin@education.gov.uk", role = "ctb"),
     person("Jake", "Tufts", , "jake.tufts@education.gov.uk", role = "ctb"),
-    person("Rich", "Bielby", , "richard.bielby@education.gov.uk", role = "ctb",
+    person("Rich", "Bielby", , "richard.bielby@education.gov.uk", role = "ctb"),
+    person( "Menna", "Zayed", , "menna.zayed@education.gov.uk", role = "ctb",
            comment = c(ORCID = "0000-0001-9070-9969"))
```
Comment (cjrace) [inline DESCRIPTION:13]: Think you've now accidentally stolen Rich's ORCID number (it's an academic thing). I think you'll want this instead:

```
    person("Rich", "Bielby", , "richard.bielby@education.gov.uk", role = "ctb",
           comment = c(ORCID = "0000-0001-9070-9969")),
    person( "Menna", "Zayed", , "menna.zayed@education.gov.uk", role = "ctb")
  )
```

There's also a `usethis::use_tidy_description()` function that's nifty and will tidy up any stray spaces / formatting in the description file for you - worth giving that a run to double check it's all 'tidy'

File: R/pretty.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (cjrace) [reply R/pretty.R]: Ah interesting! For us we always use `c` for suppressed values - there's a [list of the main data symbols we use on the analysts guide](https://dfe-analytical-services.github.io/analysts-guide/statistics-production/ud.html#data-symbols) if you're interested

**Review by cjrace — CHANGES_REQUESTED** [review]
Just one quick thing on the description left to look at

File: DESCRIPTION (hunk 2 above)
Comment (mzayeddfe) [reply DESCRIPTION:13]: my bad! all done :)

**Review by cjrace — APPROVED** [review]
Looks good - all ready to merge!

**Pull request merged**